k2/csrc/fsa_algo.cc

```diff
@@ -147,4 +147,109 @@ void Connect(const Fsa &a, Fsa *b, std::vector<int32_t> *arc_map /*=nullptr*/) {
   }
 }
 
+bool TopoSort(const Fsa &a, Fsa *b,
+              std::vector<int32_t> *state_map /*= nullptr*/) {
+  CHECK_NOTNULL(b);
+  b->arc_indexes.clear();
+  b->arcs.clear();
+
+  if (state_map) state_map->clear();
+
+  if (IsEmpty(a)) return false;
```

Comment:
maybe `true` for empty fsa? just leave `b` and `state_map` empty

Reply:
I've changed the function to return `void`.

Reply:
fixed.